### PR TITLE
Remove handling of `index.mapper.dynamic` in AutoCreateIndex

### PR DESCRIPTION
**Pull request by cwperks**
### Description

This removes the last usage of `index.mapper.dynamic` which was deprecated before the fork.

Reading the setting in this class is dead code because the settings object passed into the constructor of AutoCreateIndex is the node settings and `index.mapper.dynamic` is an index setting. 

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/12416

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/action/support/AutoCreateIndex.java [hunk 1]
```diff
@@ -109,9 +105,6 @@ public boolean shouldAutoCreate(String index, ClusterState state) {
         if (autoCreate.autoCreateIndex == false) {
             throw new IndexNotFoundException("[" + AUTO_CREATE_INDEX_SETTING.getKey() + "] is [false]", index);
         }
-        if (dynamicMappingDisabled) {
```
Comment (reta) [inline server/src/main/java/org/opensearch/action/support/AutoCreateIndex.java:112]: I think we should warn here, similarly to `MapperService`:

```
if (INDEX_MAPPER_DYNAMIC_SETTING.exists(settings)) {
            deprecationLogger.deprecate(
                index().getName() + INDEX_MAPPER_DYNAMIC_SETTING.getKey(),
                "Index [{}] has setting [{}] that is not supported in OpenSearch, its value will be ignored.",
                index().getName(),
                INDEX_MAPPER_DYNAMIC_SETTING.getKey()
            );
        }
```

Comment (cwperks) [reply server/src/main/java/org/opensearch/action/support/AutoCreateIndex.java:112]: This is dead code. The `settings` passed into the constructor are the node settings.

If you try to add `index.mapper.dynamic: false` into `opensearch.yml` the cluster fails to startup with:

```
java.lang.IllegalArgumentException: node settings must not contain any index level settings
	at org.opensearch.common.settings.SettingsModule.<init>(SettingsModule.java:172)
	at org.opensearch.node.Node.<init>(Node.java:593)
	at org.opensearch.node.Node.<init>(Node.java:419
        ...
```

Comment (reta) [reply server/src/main/java/org/opensearch/action/support/AutoCreateIndex.java:112]: Thanks @cwperks !

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
https://github.com/opensearch-project/OpenSearch/pull/13039 needs to get in first :(

**Review by reta — APPROVED** [review]

**Comment by deshsidd** [comment]
Thanks for the refactoring!

**Review by deshsidd — APPROVED** [review]

**Comment by andrross** [comment]
FYI @reta I removed the 2.14 label as it seems this has not yet been backported.

**Pull request merged**